doc/source/code_editors.rst

```diff
@@ -1,3 +1,35 @@
 ============
 Code Editors
 ============
+===================
+IDE Recommendations
+===================
+-NotepadPP
```

Comment:
honestly neither notepadpp(++), vim, nano, atom, sublime are in reality IDEs, they belong strictly to code editors (while technically speaking all support plugins that make them behave like ide)
IDEs are eclipse, xcode, intellij idea, visual studio